Dodge.js

```diff
@@ -1,178 +1,201 @@
+    this.MAXheight = Dimensions.get('window').height;
+    //this.deleteEntity = this.deleteEntity.bind(this)
+    this.addEntity(new Knight(this));
+    //this.addEntity(new Circle(this));
+    this.addEntity(new Brick(this));
+    this.addEntity(new Grenade(this));
+    this.addEntity(new Brick(this));
+    this.addEntity(new Circle(this));
+    this.addEntity(new Grenade(this));
+    this.addEntity(new Water(this))
+    setTimeout(() => {this.nav.navigate("Battle")}, 5000)
 
-const MAXwidth = Dimensions.get('window').width;
-const MAXheight = Dimensions.get('window').height;
 
+  }
 
-//adds an entity given its type, if it's a Grenade set a timer for explosion
-function AddEntity(entity){
-let id = generateUniqueId()
-ents[id] = new entity(id)
+  // this.isLetterInWord = this.isLetterInWord.bind(this);
 
-if(entity == Grenade){
-  setTimeout(() => { Explode(id) }, 2000)
-}
-}
 
-//deletes a single entity given that entities unique id
-function DeleteEntity(id){
-delete(ents[id])
-}
+  render() {
+    return (
+      <View>
+        <GameEngine
+          systems={[this.gameLoop,]}
+          var entities={this.ents}>
+          <MoveKnight />
+        </GameEngine>
+      </View>
+    );
+  }
 
-//blows up the grenade
-function Explode(id){
-  //spawns shrapnel traveling from where the grenade blew up
-  InitialShrapnel(ents[id].x,ents[id].y)
-  //delete the original projectile
-  DeleteEntity(id)
-}
+  //adds an entity given its type, if it's a Grenade set a timer for explosion
+  addEntity(entity) {
+    this.ents[entity.id] = entity
+  }
 
-//runs 60 times a second updating each entitiy
-function Gameloop(entities, { touches }, )  {
-  let toDelete = []
-  for (let id in entities) {
+  //deletes a single entity given that entities unique id
+  deleteEntity(id) {
+    delete (this.ents[id])
+    score += 10
+  }
+
+  //blows up the grenade
+  explode(id) {
+    //spawns shrapnel traveling from where the grenade blew up
+    initialShrapnel(this.ents[id].x, this.ents[id].y)
+    //delete the original projectile
+    deleteEntity(id)
+  }
+
+  //runs 60 times a second updating each entitiy
+  gameLoop(entities, { touches },) {
+    let toDelete = []
+    for (let id in entities) {
       entities[id].update()
+    }
+
+    return entities
   }
-  
-return entities
-}
 
-//creates a square component
-function CreateBrick(props) {
-return(
-  //Having a dynamic call as in props.Image produces the error
-  //Cannot find module './assets/Brick.jpeg'; which doesn't seem right
-    <Image source = {require("./assets/Brick.jpeg")}
-      style = {{        
-      position: 'absolute',
-      width: props.width,
-      height: props.height,
-      left: props.x,
-      top: props.y,
-      backgroundColor: props.backgroundColor,}}/>
-
-);
-}
-/*
-//creates a square component
-function CreateBrick(props){
-  return (
-     <View style={styles.container}>
+  //Creates a square component
+  CreateBrick(props) {
```

Comment:
I think these should probably be in "Entities.js" or maybe a new file for all the Entity rendering logic "Sprites.js" or "Renderer.js"

Reply:
There's also a lot of code-repetition here.
The easiest change is probably to have a function:
```
function sizedImageStyle(props) {
  return {
    position: 'absolte',
    width: props.width,
    ...
  }
}

Then have CreateBrick:
```
function CreateBrick(props) {
  return (
    <Image={require("...")}
        style={sizedImageStyle(props)} />
  )
}
```
But there are many ways to do this.

Reply:
Yeah so um I don't think all of those actually did anything, I moved them to entities and commented them all out and it still works fine. Dodgeball creates a new instance of one of the object classes which has all its own information as well as a reference to the game and then passes passes its information to createSprite to be rendered